mesonbuild/build.py

```diff
@@ -3003,6 +3004,143 @@ def get_classpath_args(self):
     def get_default_install_dir(self) -> T.Union[T.Tuple[str, str], T.Tuple[None, None]]:
         return self.environment.get_jar_dir(), '{jardir}'
 
+class AppBundle(Executable):
+    known_kwargs = known_nsapp_kwargs
+
+    typename = 'bundle'
+
+    # As in _CFBundleGetPlatformExecutablesSubdirectoryName (CoreFoundation)
+    _exe_dir_names_by_system = {
+        'darwin': 'MacOS',
+        'windows': 'Windows',
+        'sunos': 'Solaris',
+        # _: 'HPUX',
+        'cygwin': 'Cygwin',
+        'linux': 'Linux',
+        'freebsd': 'FreeBSD',
+        'emscripten': 'WASI',
+    }
+
+    def __init__(
+        self,
+        name: str,
+        subdir: str,
+        subproject: SubProject,
+        for_machine: MachineChoice,
+        sources: T.List['SourceOutputs'],
+        structured_sources: T.Optional[StructuredSources],
+        objects: T.List[ObjectTypes],
+        environment: environment.Environment,
+        compilers: T.Dict[str, 'Compiler'],
```

Comment:
When adding brand new code, avoid quoting annotations. Since meson moved to python 3.7 as the minimum, we started using `from __future__ import annotations` which has the equivalent effect, so the manual quoting is no longer needed -- and makes lines slightly longer, which adds up and eventually tends to cause overly long lines which then need wrapping.